### ISSUE TITLE
In what way is your server non standard?  Or why wont it connect with FileCenttipede 

### ISSUE THREAD

**Issue by PAEz** [proftpd/proftpd#1527]
### What I Did

FileCentipede Wont connect with your server.
### What I Expected/Wanted

For the app to connect.

### ProFTPD Version and Configuration

Unknown.  Its not hosted by me but by Seedbox.io so I am unable to tell.

### Explanation
I tried connecting to an ftp server running your software with FileCentipede.  It would not connect and when I asked the author of this software he said it was because your software in a non standard version of ftp......
https://github.com/filecxx/FileCentipede/issues/369
....just wondering if there is any truth in this and what the problem might be.
Thank you for your time.

**Comment by Castaglia** [comment]
Unfortunately there is not enough information, in https://github.com/filecxx/FileCentipede/issues/369 or in this ticket, to say why that cilent is having issues interacting with ProFTPD.

What would help would be any logs from that client, and the [ProFTPD debug logging](http://www.proftpd.org/docs/howto/Debugging.html) from sessions/connections from that client, to see what might be happening.

**Comment by PAEz** [comment]
Looks like the maker of FileCentipede did some tests and responded to the issue I opened with the following.  I really like FileCentipede and would love to keep using it, but I really want it to be compatible with your software to make full use of it, hopefully you guys might be able to make that happen.

>
Some non-standard implemented FTP servers will not response to the client with a 220 code when the connection is established(not ProFTPD), so FileCentipede will send a \r\n to the server (for compatibility purposes).

Then wait for the welcome message.

No matter what the remote FTP server response data is, FileCentipede will ignore it.

Then FileCentipede will send "USER username" to the server for auth.

But when ProFTPD received the \r\n message, it will respond with a message "500 Invalid command: try being more creative", when FileCentipede sends USER username*** to ProFTPD server, it also will respond with a message "500 Invalid command: try being more creative".

After that, no matter what was sent, ProFTPD will be returned in this way.

The right way is then the server received an unknown command, ignore it, or at less shouldn't die on it.
Other FTP servers won't, such as vsftpd.

So the choice is rather FileCentipede compatible with other non-standard FTP servers or ProFTPD.

**Comment by Castaglia** [comment]
OK.  I think the above assumptions (about sending a "\r\n" to the server "for compatibility purposes") may not be explicitly covered in RFCs, thus saying it's "non-standard" may not be technically accurate.  I'll try to get some time to reproduce this locally, to see what adjustments (if applicable) can be done.

**Comment by Castaglia** [comment]
@PAEz I'm not able to reproduce the reported behavior locally, using the latest ProFTPD.

In my local reproduction test, I connect to the server, and send `\r\n` before performing any other IO.  I then read the initial server banner/greeting, _and_ the expected 500 response (from the bare CRLF).  I tend send an FTP `USER` command, multiple times -- and for each `USER` command, I see the expected 331 response, _not_ the reported 500 response.

So I am left wondering just what this FileCentipede file code is doing, as I not seeing any misbehavior on ProFTPD's part...

**Comment by PAEz** [comment]
Hi, sorry for no response.
In the link I posted in the first comment (https://github.com/filecxx/FileCentipede/issues/369) I pasted a log of his app connecting to a proftpd server (I set one up) after his latest update.  It seems to connect, get past the user/pass check and then fail in some way and log out?
He doesnt seem that interested in fixing it and Ive kinda lost interest (his app has a lot of small annoying problems, shame it looks like it has a lot of promise).
Thanks for taking the time to look into it, but if hes not interested in solving it I wouldnt waste any more of you time.
